consensus/polybft/state.go

```diff
@@ -443,16 +443,22 @@ func (s *State) insertStateSyncEvent(event *types.StateSyncEvent) error {
 }
 
 // getStateSyncEventsForCommitment returns state sync events for commitment
-// if there is an event with index that can not be found in db in given range, an error is returned
-func (s *State) getStateSyncEventsForCommitment(fromIndex, toIndex uint64) ([]*types.StateSyncEvent, error) {
+// based on the getAll flag, the function will try to get all events in range, and if it fails,
+// it will either return an error, or return all the events it can without returning an error
+func (s *State) getStateSyncEventsForCommitment(fromIndex, toIndex uint64,
+	getAll bool) ([]*types.StateSyncEvent, error) {
```

Comment:
Consider returning an error in any case and handle it properly from the caller. It would imply removing `getAll` parameter entirely.

If you don't think that removing `getAll` is a good idea, then its naming is a bit misleading, since it denotes whether an error should be returned or not. In that case consider renaming it somehow (`propagateError`, not sure?)...

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1083/commits/9087bc8c572b7cb12fe8259878552b38a814f786